components/RecentlyPlayedDrawer.tsx

```diff
@@ -35,114 +35,274 @@ const PEEK_HEIGHT = 100; // How much of the drawer shows when collapsed
+      const today = new Date();
+      const easternDate = new Date(today.toLocaleString("en-US", { timeZone: "America/New_York" }));
+
+      const year = easternDate.getFullYear();
+      const month = String(easternDate.getMonth() + 1).padStart(2, "0");
+      const day = String(easternDate.getDate()).padStart(2, "0");
+
+      const dateStr = `${year}-${month}-${day}`;
+
+      const songs = await fetchShowPlaylist(currentShow, dateStr);
+      setShowPlaylists([{ showName: currentShow, songs }]);
+
+      // If current show has no songs, mark for auto-load of previous show
+      if (songs.length === 0) {
+        shouldTriggerAutoLoad = true;
+      }
     } catch (err) {
-      setError('Failed to load recently played songs');
-      debugError('Error fetching recently played:', err);
+      setError(`Failed to load playlist for ${currentShow}`);
+      debugError('Error fetching current show playlist:', err);
+      setShowPlaylists([]);
     } finally {
       if (isRefresh) {
         setRefreshing(false);
       } else {
         setLoading(false);
       }
+
+      // Trigger auto-load after loading state is cleared
+      if (shouldTriggerAutoLoad) {
+        setShouldAutoLoadPrevious(true);
+      }
     }
-  };
+  }, [currentShow, fetchShowPlaylist]);
+
+  const loadPreviousShow = useCallback(async () => {
+    // Determine which show to find the previous show for
+    const lastLoadedShow = showPlaylists.length > 0 ? showPlaylists[showPlaylists.length - 1].showName : currentShow;
+
+    if (!lastLoadedShow || !currentShow || currentShow === 'WMBR 88.1 FM' || loadingMore || hasReachedEndOfDay) {
+      return;
+    }
+
+    setLoadingMore(true);
+
+    try {
+      const scheduleService = ScheduleService.getInstance();
+      const previousShow = await scheduleService.findPreviousShow(lastLoadedShow);
+
+      if (!previousShow) {
+        setHasReachedEndOfDay(true);
+        return;
+      }
+
+      // Check if we've already loaded this specific previous show to prevent duplicates
+      const alreadyLoaded = showPlaylists.some(playlist => playlist.showName === previousShow.show.name);
+      if (alreadyLoaded) {
+        return;
+      }
+
+      try {
+        const songs = await fetchShowPlaylist(previousShow.show.name, previousShow.date);
+
+        setShowPlaylists(prev => [...prev, {
+          showName: previousShow.show.name,
+          songs
+        }]);
+      } catch (playlistError) {
+        // Handle 404 or other playlist fetch errors gracefully - still add the show with empty songs
+        setShowPlaylists(prev => [...prev, {
+          showName: previousShow.show.name,
+          songs: []
+        }]);
+      }
+    } catch (err) {
+      debugError('Error loading previous show:', err);
+    } finally {
+      setLoadingMore(false);
+    }
+  }, [currentShow, showPlaylists, loadingMore, hasReachedEndOfDay, fetchShowPlaylist]);
+
+  // Clear playlist data when current show changes
```

Comment:
IMO this is a ton of churn. we should be able to not clear this and instead just clear if we change days - if we're going forward in time we're just adding new records, not removing. would reduce tons of api calls on the hour -> two api calls (show list + songs).

or possibly doesn't need to refetch at all, since we'd have the songs?